openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -179,134 +179,193 @@ def check_permissions_for_scope(
-        else:
-            a_user = self.datastore.get(
-                fqid_from_collection_and_id("user", instance_id),
-                ["meeting_ids"],
-                lock_result=False,
+        elif not (
+            a_meeting_ids := set(
+                self.datastore.get(
+                    fqid_from_collection_and_id("user", self.user_id),
+                    ["meeting_ids"],
+                    lock_result=False,
+                ).get("meeting_ids", [])
             )
-            a_meeting_ids = set(a_user.get("meeting_ids", []))
-        if not a_meeting_ids:
+        ):
             return False
         intersection_meeting_ids = a_meeting_ids.intersection(b_meeting_ids)
         if not b_meeting_ids.issubset(intersection_meeting_ids):
             return False
-        intersection_meetings = self.datastore.get_many(
+        return self.datastore.get_many(
             [
                 GetManyRequest(
                     "meeting",
                     list(intersection_meeting_ids),
-                    ["admin_group_id", "group_ids", "meeting_user_ids"],
+                    ["admin_group_id", "group_ids"],
                 )
             ],
             lock_result=False,
         ).get("meeting", {})
-        for meeting_id, meeting_dict in intersection_meetings.items():
-            admin_meeting_users = {}
-            admin_group_id = meeting_dict.get("admin_group_id", 0)
-            admin_group = self.datastore.get(
-                fqid_from_collection_and_id("group", admin_group_id),
-                ["meeting_user_ids"],
-                lock_result=False,
-            )
-            admin_meeting_users = self.datastore.get_many(
+
+    def _collect_admin_meeting_users(
+        self, intersection_meetings: dict[int, Any]
+    ) -> set[int]:
+        """
+        Gets the admin groups and those groups with permission User.CAN_UPDATE and USER.CAN_MANAGE of the meetings.
+        Returns a set of the groups meeting_user_ids.
+        """
+        group_ids = [
+            group_id
+            for meeting_id, meeting_dict in intersection_meetings.items()
+            for group_id in meeting_dict.get("group_ids", [])
+        ]
+        admin_group_ids = [
+            meeting_dict["admin_group_id"]
+            for meeting_dict in intersection_meetings.values()
+        ]
+        return {
+            *{
+                mu_id
+                for group_id, group in self.datastore.get_many(
+                    [
+                        GetManyRequest(
+                            "group", group_ids, ["meeting_user_ids", "permissions"]
+                        )
+                    ],
+                    lock_result=False,
+                )
+                .get("group", {})
+                .items()
+                if (
+                    "user.can_update" in group.get("permissions", [])
+                    or "user.can_manage" in group.get("permissions", [])
```

Comment:
Additionally check if "admin_group_for_meeting_id" is set and you can skip the other comprehension and the variable `admin_group_ids`